web/react/components/textbox.jsx

```diff
@@ -279,7 +292,16 @@ export default class Textbox extends React.Component {
         this.doProcessMentions = true;
     }
 
+    showPreview(e) {
+        e.preventDefault();
+        e.target.blur();
+        this.setState({preview: !this.state.preview});
+        this.resize();
+    }
+
     render() {
+        let previewLinkVisible = (this.value || '').length > 0;
```

Comment:
I wasn't quite sure if `this.props.messageText` actually gets updated with each keystroke since as far as I understand props can only be set from the parent component (though I may be wrong here, pretty new to react).
I'll check that ;)